static/js/user_profile.js

```diff
@@ -192,6 +214,62 @@ export function register_click_handlers() {
         e.preventDefault();
     });
 
+    $("body").on("submit", "#user-profile-modal .subscription_list_remove form", (e) => {
```

Comment:
Can this just be an on-click handler?  We've been moving away from using `submit` handlers because it can come with unexpected implicit behavior UI for an app like Zulip (like `Enter` automatically submitting the form).  And I think that's especially appropriate for a widget like this, which isn't a form.